src/Microsoft.DotNet.XHarness.CLI/Android/AndroidTestCommand.cs

```diff
@@ -1,67 +1,121 @@
+            _log.LogDebug("Arguments to instrumentation:");
+
+            if (File.Exists(_arguments.AppPackagePath))
+            {
+                var runner = new AdbRunner(_log);
+                string apkName = Path.GetFileNameWithoutExtension(_arguments.AppPackagePath);
+
+                try
+                {
+                    using (_log.BeginScope("Initialization and setup of APK on device"))
+                    {
+                        runner.KillAdbServer();
+                        runner.StartAdbServer();
+                        runner.ClearAdbLog();
+
+                        _log.LogDebug($"Working with {runner.GetAdbVersion()}");
+
+                        // If anything changed about the app, Install will fail; uninstall it first.
+                        // (we'll ignore if it's not present)
+                        runner.UninstallApk(apkName);
+                        if (runner.InstallApk(_arguments.AppPackagePath) != 0)
+                        {
+                            _log.LogCritical("Install failure: Test command cannot continue");
+                            return Task.FromResult(-12);
+                        }
+                        runner.KillApk(apkName);
+
+                        // App needs to be able to read and write the storage folders we use for IO
+                        runner.GrantPermissions(apkName,
+                            new string[] {"android.permission.READ_EXTERNAL_STORAGE",
+                                          "android.permission.WRITE_EXTERNAL_STORAGE"});
+                    }
+
+                    // No class name = default Instrumentation
+                    runner.RunApkInstrumentation(apkName, _arguments.InstrumentationName, _arguments.InstrumentationArguments);
+
+                    using (_log.BeginScope("Post-test copy and cleanup"))
+                    {
+                        var logs = runner.PullFiles("/sdcard/Documents/helix-results", _arguments.OutputDirectory);
+                        foreach (string log in logs)
+                        {
+                            _log.LogDebug($"Detected output file: {log}");
+                        }
+                        runner.DumpAdbLog(Path.Combine(_arguments.OutputDirectory, "adb-logcat.log"));
+                        runner.UninstallApk(apkName);
+                        runner.KillAdbServer();
+                    }
+
+                    return Task.FromResult(0);
+                }
+                catch (Exception toLog)
+                {
+                    _log.LogCritical(toLog, "Failure to run test package");
+                }
+                finally
+                {
+                    runner.KillAdbServer();
+                }
+            }
+            else
+            {
+                _log.LogCritical($"Couldn't find {_arguments.AppPackagePath}!");
+                return Task.FromResult(-42);

```

Comment:
We are returning quite a lot of magic numbers from many places all over the code (failed arg validation, bad help and so on..).
Maybe we should do the right thing and create a CLI project wide enum with list of result codes so that we are uniform and have a good overview of what can come out of the tool?

Reply:
Yes, sharing the error codes between iOS and Android makes sense, @MattGal can you create an enum and we will adapt to it?

Reply:
sure thing.  This is part of the WIP, I just want unique failure stuff.

Reply:
added